lib/ansible/module_utils/common/dict_transformations.py

```diff
@@ -121,3 +121,21 @@ def dict_merge(a, b):
         else:
             result[k] = deepcopy(v)
     return result
+
+
+def recursive_diff(dict1, dict2):
```

Comment:
Looks good from a cloudfront_distribution point of view. +1 to moving into common code, so we can use in AWS modules. Thanks!